datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -456,3 +402,73 @@ def should_bypass_proxy(url, no_proxy_uris):
             if no_proxy_uri == parsed_uri or parsed_uri.endswith(dot_no_proxy_uri):
                 return True
     return False
+
+
+def create_basic_auth(config):
+    # Since this is the default case, only activate when all fields are explicitly set
+    if config['username'] and config['password']:
+        return (config['username'], config['password'])
+
+
+def create_digest_auth(config):
+    return requests_auth.HTTPDigestAuth(config['username'], config['password'])
+
+
+def create_ntlm_auth(config):
+    global requests_ntlm
+    if requests_ntlm is None:
+        import requests_ntlm
+
+    return requests_ntlm.HttpNtlmAuth(config['ntlm_domain'], config['password'])
+
+
+def create_kerberos_auth(config):
+    global requests_kerberos
+    if requests_kerberos is None:
+        import requests_kerberos
+
+        KERBEROS_STRATEGIES['required'] = requests_kerberos.REQUIRED
+        KERBEROS_STRATEGIES['optional'] = requests_kerberos.OPTIONAL
+        KERBEROS_STRATEGIES['disabled'] = requests_kerberos.DISABLED
+
+    # For convenience
+    if is_affirmative(config['kerberos_auth']):
+        config['kerberos_auth'] = 'required'
+
+    if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
+        raise ConfigurationError(
+            'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
+                config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
+            )
+        )
+
+    return requests_kerberos.HTTPKerberosAuth(
+        mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
+        delegate=is_affirmative(config['kerberos_delegate']),
+        force_preemptive=is_affirmative(config['kerberos_force_initiate']),
+        hostname_override=config['kerberos_hostname'],
+        principal=config['kerberos_principal'],
+    )
+
+
+def create_aws_auth(config):
+    global requests_aws
+    if requests_aws is None:
+        from aws_requests_auth import boto_utils as requests_aws
+
+    for setting in ('aws_host', 'aws_region', 'aws_service'):
+        if not config[setting]:
+            raise ConfigurationError('AWS auth requires the setting `{}`'.format(setting))
+
+    return requests_aws.BotoAWSRequestsAuth(
+        aws_host=config['aws_host'], aws_region=config['aws_region'], aws_service=config['aws_service']
+    )
+
+
+AUTH_TYPES = {
+    'basic': create_basic_auth,
+    'digest': create_digest_auth,
+    'ntlm': create_ntlm_auth,
+    'kerberos': create_kerberos_auth,
+    'aws': create_aws_auth,
```

Comment:
RequetsWrapper is now quite critical, so I think we need a test for each auth_type.

Reply:
We do, what do you mean?